### PR TITLE
add new target EMAX_F405_ELRS_AIO

### PR DESCRIPTION
**Pull request by EmaxModel**
I reworked the previous error and PR form a branch to the master.Thanks for the correction.

### CODE REVIEW

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+#pragma once
+
+#define FC_TARGET_MCU       STM32F405
+#define BOARD_NAME          EMAX_F405_ELRS_AIO
+#define MANUFACTURER_ID     EMAX 
+
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM42688P 
+#define USE_ACC_SPI_ICM42688P
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:30]: Need `#define USE_ACC`

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+#define LED1_PIN        PB5
+#define SPI1_SCK_PIN    PA5
+#define SPI2_SCK_PIN    PB13
+#define SPI3_SCK_PIN    PC10
+#define SPI1_SDI_PIN    PA6
+#define SPI2_SDI_PIN    PB14
+#define SPI3_SDI_PIN    PC11
+#define SPI1_SDO_PIN    PA7
+#define SPI2_SDO_PIN    PB15
+#define SPI3_SDO_PIN    C12
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:60]: `PC12`

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+#define UART3_TX_PIN    PB10
+#define UART4_TX_PIN    PA0
+#define UART6_TX_PIN    PC6
+#define UART1_RX_PIN    PA10
+#define UART3_RX_PIN    PB11
+#define UART4_RX_PIN    PA1
+#define UART5_RX_PIN    PD2
+#define UART6_RX_PIN    PC7
+#define INVERTER3_PIN   PC9
+#define INVERTER6_PIN   PC8
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:50]: Are there really two hardware inverters on the FC?

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+#define ADC_RSSI_PIN    PC0
+#define ADC_CURR_PIN    PC1
+#define FLASH_CS_PIN    PB12
+#define MAX7456_CS_PIN  PA15
+#define GYRO_1_EXTI_PIN PC4
+#define GYRO_1_CS_PIN   PA4
+#define USB_DETECT_PIN  PC5
+
+#define TIMER_PIN_MAPPING \
+    TIMER_PIN_MAP(0,PB8 , 2, -1) \
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:73]: Why does ESC_SERIAL have a timer assignment?

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    TIMER_PIN_MAP(1,PC8 , 2, -1) \
+    TIMER_PIN_MAP(2,PB0 , 2, 0) \
+    TIMER_PIN_MAP(3,PB1 , 2, 0) \
+    TIMER_PIN_MAP(4,PA3 , 1, 1) \
+    TIMER_PIN_MAP(5,PA2 , 1, 0) \
+    TIMER_PIN_MAP(6,PB6 , 1, 0) \
+    TIMER_PIN_MAP(7,PA8 , 1, -1) \
+    TIMER_PIN_MAP(8,PA9 , 1, 0) \
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:81]: Why do these pins have timer assignments?

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    TIMER_PIN_MAP(9,PA10 , 1, 0)
+
+#define ADC2_DMA_OPT        1
+#define SPI2_TX_DMA_OPT     0
+
+
+#define SERIALRX_PROVIDER CRSF
+#define MAG_BUSTYPE SPI
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:89]: There is no mag defined

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+#define DEFAULT_BARO_DEVICE BARO_NONE
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:90]: Not needed

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+#define BLACKBOX_DEVICE SPIFLASH
+#define DSHOT_BURST ON
+#define CURRENT_METER ADC
+#define BATTERY_METER ADC
+#define BEEPER_INVERSION ON
+#define BEEPER_OD OFF
+#define SYSTEM_HSE_MHZ 8
+#define MAX7456_SPI_BUS SPI3
+#define FLASH_SPI_BUS SPI2
+#define GYRO_1_BUSTYPE SPI
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:101]: Should be `#define GYRO_1_SPI_INSTANCE SPI1`

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+#define DEFAULT_BARO_DEVICE BARO_NONE
+#define ADC_DEVICE 2
+#define BLACKBOX_DEVICE SPIFLASH
+#define DSHOT_BURST ON
+#define CURRENT_METER ADC
+#define BATTERY_METER ADC
+#define BEEPER_INVERSION ON
+#define BEEPER_OD OFF
+#define SYSTEM_HSE_MHZ 8
+#define MAX7456_SPI_BUS SPI3
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:99]: Should be `#define MAX7456_SPI_INSTANCE SPI3`

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+#define FLASH_SPI_BUS SPI2
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:100]: Should be `#define FLASH_SPI_INSTANCE SPI2`

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+#define SPI1_SCK_PIN    PA5
+#define SPI2_SCK_PIN    PB13
+#define SPI3_SCK_PIN    PC10
+#define SPI1_SDI_PIN    PA6
+#define SPI2_SDI_PIN    PB14
+#define SPI3_SDI_PIN    PC11
+#define SPI1_SDO_PIN    PA7
+#define SPI2_SDO_PIN    PB15
+#define SPI3_SDO_PIN    C12
+#define BARO_CS_PIN     PB3
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:61]: Should not be defined if there is no SPI baro

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+#define BATTERY_METER ADC
+#define BEEPER_INVERSION ON
+#define BEEPER_OD OFF
+#define SYSTEM_HSE_MHZ 8
+#define MAX7456_SPI_BUS SPI3
+#define FLASH_SPI_BUS SPI2
+#define GYRO_1_BUSTYPE SPI
+#define GYRO_1_SPIBUS SPI1
+#define GYRO_1_SENSOR_ALIGN CW90
+#define GYRO_1_ALIGN_YAW 900
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:104]: Should be `#define GYRO_1_ALIGN CW90_DEG`

**Review by ot0tot — CHANGES_REQUESTED** [review]
Refer to another config for the defines. Do you have a wiring diagram for this FC?

**Comment by EmaxModel** [comment]
Thanks for correcting my error, I submitted the new target

**Comment by ot0tot** [comment]
There are additional invalid defines that should be removed. Have you tested this config yet? 

I think DSHOT_BURST ON should be removed as well since burst DMA is incompatible with bidirectional DShot on F4 MCUs.

**Comment by EmaxModel** [comment]
This configuration has been tested，Can I change it to DSHOT BURST AUTO？If removed, the motor will not be recognized

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+#define FLASH_CS_PIN    PB12
+#define MAX7456_CS_PIN  PA15
+#define GYRO_1_EXTI_PIN PC4
+#define GYRO_1_CS_PIN   PA4
+#define USB_DETECT_PIN  PC5
+
+#define TIMER_PIN_MAPPING \
+    TIMER_PIN_MAP(0,PB0 , 2, 0) \
+    TIMER_PIN_MAP(1,PB1 , 2, 0) \
+    TIMER_PIN_MAP(2,PA3 , 1, 1) \
```
Comment (ot0tot) [inline configs/EMAX_F405_ELRS_AIO/config.h:71]: This should be ` TIMER_PIN_MAP(2,PA3 , 1, 0) \` to not conflict with the DMA assignment for Motor 1

Comment (EmaxModel) [reply configs/EMAX_F405_ELRS_AIO/config.h]: thanks

**Comment by EmaxModel** [comment]
@ot0tot Sorry for the late reply, I have changed the conversations question，Can it be merge？

**Review by nerdCopter — APPROVED** [review]
- request second review

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+
+#define FC_TARGET_MCU       STM32F405
+#define BOARD_NAME          EMAX_F405_ELRS_AIO
+#define MANUFACTURER_ID     EMAX 
+
+#define USE_ACC
+#define USE_GYRO
+#define USE_GYRO_SPI_ICM42688P 
+#define USE_ACC_SPI_ICM42688P 
+
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:32]:
```suggestion
#define USE_MAX7456
#define USE_FLASH
#define USE_FLASH_W25Q128FV

```

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 15]
```diff
@@ -0,0 +1,97 @@
...
+    TIMER_PIN_MAP(2,PA3 , 1, 0) \
+    TIMER_PIN_MAP(3,PA2 , 1, 0) \
+    TIMER_PIN_MAP(4,PA10 , 1, 0)
+
+#define ADC2_DMA_OPT        1
+#define SPI2_TX_DMA_OPT     0
+
+
+#define SERIALRX_PROVIDER CRSF
+#define ADC_DEVICE 2
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:80]:
```suggestion
#define ADC_INSTANCE ADC2
```

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 16]
```diff
@@ -0,0 +1,97 @@
...
+#define BLACKBOX_DEVICE SPIFLASH
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:81]:
```suggestion
#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
```

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 17]
```diff
@@ -0,0 +1,97 @@
...
+#define DSHOT_BURST AUTO
+#define CURRENT_METER ADC
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:83]:
```suggestion
#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
```

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+#define BATTERY_METER ADC
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:84]:
```suggestion
#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
```

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+#define SYSTEM_HSE_MHZ 8
+#define MAX7456_SPI_INSTANCE SPI3
+#define FLASH_SPI_INSTANCE SPI2
+#define GYRO_1_SPI_INSTANCE SPI1
+#define GYRO_1_SPIBUS SPI1
+#define GYRO_1_ALIGN CW90_DEG
+#define GYRO_1_ALIGN_YAW 900
+#define GYRO_2_BUSTYPE SPI
+#define GYRO_2_SPIBUS 1
+#define GYRO_1_SENSOR_ALIGN CW0
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:96]:
```suggestion
#define GYRO_1_SENSOR_ALIGN CW0_DEG
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]
Missing flash driver and OSD driver.

**Comment by EmaxModel** [comment]
Thank you for your correction

File: configs/EMAX_F405_ELRS_AIO/config.h (hunk 5 above)
Comment (nerdCopter) [reply configs/EMAX_F405_ELRS_AIO/config.h]: if this is resolved, please click `Resolve conversation` button

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 20]
```diff
@@ -0,0 +1,100 @@
...
+#define BEEPER_OD OFF
+#define SYSTEM_HSE_MHZ 8
+#define MAX7456_SPI_INSTANCE SPI3
+#define FLASH_SPI_INSTANCE SPI2
+#define GYRO_1_SPI_INSTANCE SPI1
+#define GYRO_1_SPIBUS SPI1
+#define GYRO_1_ALIGN CW90_DEG
+#define GYRO_1_ALIGN_YAW 900
+#define GYRO_2_BUSTYPE SPI
+#define GYRO_2_SPIBUS 1
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:98]: There is no GYRO_2
```suggestion
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]
Found two more :)

File: configs/EMAX_F405_ELRS_AIO/config.h [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+    TIMER_PIN_MAP(4,PA10 , 1, 0)
+
+#define ADC2_DMA_OPT        1
+#define SPI2_TX_DMA_OPT     0
+
+
+#define SERIALRX_PROVIDER CRSF
+#define ADC_INSTANCE ADC2
+#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
+#define DSHOT_BURST AUTO
```
Comment (haslinghuis) [inline configs/EMAX_F405_ELRS_AIO/config.h:85]: This define does not exist.

```suggestion
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

**Review by haslinghuis — APPROVED** [review]
Should be okay now.

**Review by nerdCopter — APPROVED** [review]

File: configs/EMAX_F405_ELRS_AIO/config.h (hunk 6 above)
Comment (haslinghuis) [reply configs/EMAX_F405_ELRS_AIO/config.h]: solved

File: configs/EMAX_F405_ELRS_AIO/config.h (hunk 7 above)
Comment (haslinghuis) [reply configs/EMAX_F405_ELRS_AIO/config.h]: solved

File: configs/EMAX_F405_ELRS_AIO/config.h (hunk 5 above)
Comment (haslinghuis) [reply configs/EMAX_F405_ELRS_AIO/config.h]: Resolved

**Pull request merged**